### PR TITLE
chore(servicecatalog): cleanup unit tests for TagOptions

### PR DESCRIPTION
**Pull request by arcrank**
We implemented TagOptions as a full construct, and created its own unit test suite. We are moving
some of the basic validation tests out of the other resources unit tests, and then the cross resource association
tests as well.  The only TagOption tests that remain in portfolio/product are for testing the association and adding
as a prop, validation and multi resource tests will be in the `tag-option` test suite.

The removed validation tests are already in the `tag-option` test suite. 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts [hunk 1]
```diff
@@ -350,44 +350,6 @@ describe('portfolio associations and product constraints', () => {
     Template.fromStack(stack).resourceCountIs('AWS::ServiceCatalog::TagOptionAssociation', 3);
   }),
 
-  test('fails to create and then add tag options with invalid minimum key length', () => {
```
Comment (arcrank) [inline packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts:353]: see
https://github.com/aws/aws-cdk/blob/9940952d67bdf07f3d737dc88676dc7f7c435a12/packages/%40aws-cdk/aws-servicecatalog/test/tag-option.test.ts#L26-L34

File: packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts [hunk 2]
```diff
@@ -350,44 +350,6 @@ describe('portfolio associations and product constraints', () => {
...
-  });
-
-  test('fails to create and then add tag options with invalid maxium key length', () => {
-    expect(() => {
-      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
-        allowedValuesForTags: {
-          ['key1'.repeat(1000)]: ['value1', 'value2'],
-          key2: ['value1'],
-        },
-      });
```
Comment (arcrank) [inline packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts:373]: see https://github.com/aws/aws-cdk/blob/9940952d67bdf07f3d737dc88676dc7f7c435a12/packages/%40aws-cdk/aws-servicecatalog/test/tag-option.test.ts#L36-L44

File: packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts [hunk 3]
```diff
@@ -350,44 +350,6 @@ describe('portfolio associations and product constraints', () => {
...
-  test('fails to create and then add tag options with invalid value length', () => {
-    expect(() => {
-      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
-        allowedValuesForTags: {
-          key1: ['value1'.repeat(1000), 'value2'],
-          key2: ['value1'],
-        },
-      });
-      portfolio.associateTagOptions(tagOptions);
-    }).toThrowError(/Invalid TagOption value for resource/);
```
Comment (arcrank) [inline packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts:388]: see https://github.com/aws/aws-cdk/blob/9940952d67bdf07f3d737dc88676dc7f7c435a12/packages/%40aws-cdk/aws-servicecatalog/test/tag-option.test.ts#L46-L54

**Review by skinny85 — APPROVED** [review]
Nice! Much tidier and easier to review than putting everything into one PR. I hope you agree 🙂.

**Pull request merged**